astropy/utils/masked/core.py

```diff
@@ -914,6 +912,8 @@ def _masked_result(self, result, mask, out):
             # Quantity), so use Masked to determine the appropriate class.
             return Masked(result, mask)
 
+        if not isinstance(out, Masked):
+            out = Masked(out, np.isnan(out))
```

Comment:
Is this to address the comment below here?

Reply:
I really do not understand this addition - this is for wrapping the result of `__array_function__` or `__array_ufunc__` with input that is `Masked` - really the `assert` should never get hit!